engine/src/main/java/org/terasology/engine/rendering/nui/internal/LwjglCanvasRenderer.java

```diff
@@ -86,7 +86,7 @@
     private DisplayDevice displayDevice;
     private float uiScale = 1f;
 
-    private Matrix4fStack modelMatrixStack = new Matrix4fStack(1000);
+    private Matrix4fStack modelMatrixStack = new Matrix4fStack(10000);
```

Comment:
two magic numbers :eyes: what is this change good for?

Reply:
Matrix4fStack is preallocated so the matrices are reused. we need to allocate something ahead of time.